src/tribler/gui/network/request.py

```diff
@@ -100,57 +100,64 @@ def _set_url(self, base_url: str):
                           for key, value in self.url_params.items()}
             self.url += '?' + urlencode(url_params, doseq=True)
 
-    def update_status(self, status_code: int):
-        self.logger.debug(f'Update {self}: {status_code}')
-        self.status_code = status_code
-        if status_code > 0:  # positive codes are HTTP response codes
-            self.status_text = str(status_code)
-        else:  # negative codes represent QNetworkReply error codes
-            self.status_text = f'{status_code}: {reply_errors.get(-status_code, "<unknown error code>")}'
-
     def on_finished(self):
         if not self.reply or not self.manager:
             return
 
         self.logger.info(f'Finished: {self}')
         try:
-            error_code = self.reply.error()
-            if error_code != QNetworkReply.NoError:
-                error_name = reply_errors.get(error_code, '<unknown error>')
-                self.logger.warning(f'Request {self} finished with error: {error_code} ({error_name})')
-                self.update_status(-error_code)
-                return
-
+            # If HTTP status code is available on the reply, we process that first.
+            # This is because self.reply.error() is not always QNetworkReply.NoError even if there is HTTP response.
+            # One example case is for HTTP Status Code 413 (HTTPRequestEntityTooLarge) for which QNetworkReply
+            # error code is QNetworkReply.UnknownContentError
             if status_code := self.reply.attribute(QNetworkRequest.HttpStatusCodeAttribute):
-                self.update_status(status_code)
-
-            data = bytes(self.reply.readAll())
-            if self.raw_response:
-                self.logger.debug('Create a raw response')
-                header = self.reply.header(QNetworkRequest.ContentTypeHeader)
-                self.on_finished_signal.emit((data, header))
+                self._handle_http_response(status_code)
                 return
 
-            if not data:
-                self.logger.error(f'No data received in the reply for {self}')
-                return
-
-            self.logger.debug('Create a json response')
-            result = json.loads(data)
-            if isinstance(result, dict):
-                result[REQUEST_ID] = self.id
-            is_error = 'error' in result
-            if is_error and self.capture_errors:
-                text = self.manager.show_error(self, result)
-                raise Warning(text)
+            # Process any other NetworkReply Error response.
+            error_code = self.reply.error()
+            if error_code != QNetworkReply.NoError:
```

Comment:
The `if` here looks incorrect. If it is possible that the condition is not satisfied (that is, the error code is `QNetworkReply.NoError`), the reply is not handled in this case.

Probably, it is impossible to have the situation when the HTTP code is not set, and the `QNetworkReply` error is `NoError`. So, `_handle_network_reply_errors` can be called unconditionally here.